### ISSUE TITLE
TypeError: must be string, not float, export error

### ISSUE THREAD

**Issue by ukanga** [onaio/onadata#834]
```python
Exception in request: TypeError: must be string, not floatTraceback (most recent call last):
  File "/.../onadata/apps/viewer/tasks.py", line 115, in create_xls_export
    options
  File "/.../onadata/libs/utils/export_tools.py", line 175, in generate_export
    options=options, columns_with_hxl=columns_with_hxl
  File "/.../onadata/libs/utils/export_builder.py", line 671, in to_xls_export
    self.pre_process_row(row, section),
  File "/.../onadata/libs/utils/export_builder.py", line 470, in pre_process_row
    value, elm['type'])
  File "/.../onadata/libs/utils/export_builder.py", line 437, in convert_type
    return func(value)
  File "/.../onadata/libs/utils/export_builder.py", line 190, in <lambda>
    'date': lambda x: ExportBuilder.string_to_date_with_xls_validation(x),
  File "/.../onadata/libs/utils/export_builder.py", line 201, in string_to_date_with_xls_validation
    date_obj = datetime.strptime(date_str, '%Y-%m-%d').date()
TypeError: must be string, not float
```

**Comment by faith-mutua** [comment]
This also affects Google Sheets export 

![screenshot_110216_115526_am](https://cloud.githubusercontent.com/assets/17847885/19922338/40b26634-a0f3-11e6-829f-d40273bcfd71.jpg)

**Comment by ukanga** [comment]
closed by #835
